src/solana/index.ts

```diff
@@ -65,48 +95,48 @@ export class Solana extends Blockchain {
   public async createAccount(
     address: PublicKey,
     lamports: number,
+    source: string | null,
+    lockup: Lockup | null = Lockup.default,
   ): Promise<ApiResponse<CreateAccountResponse>> {
     // Check if the amount is greater than or equal to the minimum amount
-    if (lamports < MIN_AMOUNT * LAMPORTS_PER_SOL) {
+    if (lamports < MIN_AMOUNT) {
       this.throwError('MIN_AMOUNT_ERROR', MIN_AMOUNT.toString());
     }
 
     try {
       const publicKey = new PublicKey(address);
-      const stakeAccount = Keypair.generate();
 
       // Get the minimum balance for rent exemption
       const minimumRent =
         await this.connection.getMinimumBalanceForRentExemption(
           StakeProgram.space,
         );
-
-      const amountToStake = minimumRent + lamports;
-
-      // Create the stake account
-      const createStakeAccountTx = new Transaction().add(
-        StakeProgram.createAccount({
-          authorized: new Authorized(publicKey, publicKey),
-          fromPubkey: publicKey,
-          lamports: amountToStake,
-          stakePubkey: stakeAccount.publicKey,
-        }),
-      );
-
-      const blockhash = await this.getBlockhash();
-      createStakeAccountTx.recentBlockhash = blockhash;
-
-      const createStakeAccountVerTx = await this.prepareTransaction(
+      lockup = lockup === null ? Lockup.default : lockup;
+
+      const [createStakeAccountTx, stakeAccountPublicKey, externalSigners] =
+        source === null
+          ? await this.createAccountTx(
+              publicKey,
+              lamports + minimumRent,
```

Comment:
if we decided to use bigNumber internally in the SDK then we should avoid such operations.  
`BigNumber(lamports).plus(minimumRent)`
The same with `lamports < MIN_AMOUNT` and so on 
Not a blocker.

Reply:
i think there is sense to use BigNumber when we need more precision, like multiply two numbers. Because we get it as number type  from params and then pass it into web3 lib also as number type. However we have legacy code which was transferred into new version like a legacy and it uses BiNumber (we can discuss it together the next day)